### ISSUE TITLE
[Indigo] Deprecation Warning CostMap2D Parameter Layout

### ISSUE THREAD

**Issue by fmessmer** [ipa320/cob_robots#231]
While hunting for bugs, deprecation warnings and other things in `indigo_dev`, I found that with our current parameter layout in the `local_costmap_params.yaml` files we have the same issue as reported here:
http://answers.ros.org/question/83031/move_base-and-costmap_layers/

There is a response to the thread which explains what should be done.
The issue is also documented on the hydro migration guide:
http://wiki.ros.org/hydro/Migration#Navigation_-_Costmap2D

As I'm not working with navigation, I'm not sure whether the deprecation warning 

```
[ INFO] [1413985288.872836730, 3.882000000]: Loading from pre-hydro parameter style
[ INFO] [1413985288.946924195, 3.918000000]: Using plugin "obstacle_layer"
[ INFO] [1413985289.094331450, 3.957000000]:     Subscribed to Topics: laser_scan_sensor_front laser_scan_sensor_rear
```

already results in false behaviour or not.

However, I wanted to document it!
@abubeck @ipa-mig @ipa-frm @ipa-srd @ipa-flg

**Comment by mgruhler** [comment]
@ipa-fxm Thanks, we need to change this at some time... The result currently is compatible with the groovy one. But we should start to get rid of those warnings...

**Comment by fmessmer** [comment]
@ipa-mig:
ping

**Comment by fmessmer** [comment]
@ipa-srd-fg
While you investigate the costmap node and configuration for multi-robot support, could you take care of this at the same time please? Don't know whether this actually still exists...@ipa-mig?

**Comment by ipa-srd-fg** [comment]
Sure.
Furthermore, I still get the deprecation warnings from above.

**Comment by mgruhler** [comment]
@ipa-srd-fg perfect, thanks.

I actually already tried to start it last week, but didn't get around to make progress...
Best drop by my office tomorrow so that we can discuss how to proceed best.

What needs to be done to fix this issue:
- Set up new parameter style for one robot (best take a r@w, you can find a working example in ipa_navigation/ipa_navigation_config)
- test in simulation
- test on hardware
- port to all other robots
- test on Cob

**Comment by mgruhler** [comment]
@ipa-srd-fg as discussed, you can find a draft of the adapted structure on my remote, branch `fix/deprecated_costmap_params` ([Link](https://github.com/ipa-mig/cob_navigation/tree/fix/deprecated_costmap_params)). Currently only for `r@w3-2`.

Can you check if this works?

**Comment by ipa-srd-fg** [comment]
As discussed with @ipa-mig, the deprecation warning is not only an issue for `local_costmap_params.yaml`, but also for the costmaps of the ipa_navigation repositories. Since I have not had much to do with the costmap node, I am currently not that familiar with the new parameter style. Therefore, I would like to postpone this issue until I have finished the multi robot support in order to tackle one problem after the other, if that is ok.

**Comment by fmessmer** [comment]
I'm fine with that as long as we remember this issue and eventually solve it :wink:

**Comment by mgruhler** [comment]
Not forgotten :-) I have a draft (for navigation) on [a feature branch](https://github.com/ipa-mig/cob_navigation/tree/fix/deprecated_costmap_params) which seems to work in simulation.
But we need to test this further and also on hardware. As it is for `r@w3-2`, testing it on another `r@w` make sense.

Once those tests are complete, we "just" have to port it to the other robots...

**Comment by mgruhler** [comment]
fixed in #470 and ipa320/cob_navigation#66
